.github/workflows/build.yml

```diff
@@ -28,15 +28,13 @@ jobs:
       run: ./scripts/ci/check-cabal-files.sh
 
   build:
-    runs-on: ${{ matrix.os }}
+    runs-on: windows-latest
 
-    strategy:
-      fail-fast: false
-      matrix:
-        ghc: ["8.10.7", "9.2.5"]
-        os: [ubuntu-latest, windows-latest]
+    if: ${{ github.event_name == 'push' || !github.event.pull_request.draft }}
 
     env:
+
+      ghc: "9.2.5"
       # Modify this value to "invalidate" the cabal cache.
       CABAL_CACHE_VERSION: "2022-12-30"
```

Comment:
I think it's better to still use `matrix` even if we just have a single entry for `ghc`.  In the future we might again have a period of time when we will need to support two versions of `ghc`.